apps/openmw/mwworld/cellstore.cpp

```diff
@@ -1024,4 +1061,83 @@ namespace MWWorld
             }
         }
     }
+
+    void MWWorld::CellStore::rechargeItems(float duration)
+    {
+        if (!mRechargingItemsUpToDate)
+        {
+            updateRechargingItems();
+            mRechargingItemsUpToDate = true;
+        }
+        for (TRechargingItems::iterator it = mRechargingItems.begin(); it != mRechargingItems.end(); ++it)
+        {
+            Ptr ptr = getCurrentPtr(it->first);
+
+            if (ptr.getCellRef().getEnchantmentCharge() == -1
+                    || ptr.getCellRef().getEnchantmentCharge() == it->second)
+                continue;
+
+            static float fMagicItemRechargePerSecond = MWBase::Environment::get().getWorld()->getStore().get<ESM::GameSetting>().find(
+                        "fMagicItemRechargePerSecond")->mValue.getFloat();
+
+            if (ptr.getCellRef().getEnchantmentCharge() <= it->second)
```

Comment:
This comparison is always true. Or at least the "equal" part of the "less or equal" is redundant.

Reply:
I just did not touch the logic - we use the same check now.